exasol/error/_error.py

```diff
@@ -66,33 +107,42 @@ def ExaError(
 
     Returns:
         An error type containing all relevant information.
-
-
-    Raises:
-        This function should not raise under any circumstances.
-        In case of error, it should report it also as an error type which is returned by this function.
-        Still the returned error should contain a references or information about the original call.
-
-        Attention:
-
-            FIXME: Due to legacy reasons this function currently still may raise an `ValueError` (Refactoring Required).
-
-            Potential error scenarios which should taken into account are the following ones:
-
-             * E-ERP-1: Invalid error code provided
-                    params:
-                        * Original ErrorCode
-                        * Original error message
-                        * Original mitigations
-                        * Original parameters
-             * E-ERP-2 Unknown error/exception occurred
-                    params:
-                        * exception/msg
-                        * Original ErrorCode
-                        * Original error message
-                        * Original mitigations
-                        * Original parameters
-
-            see also [Github Issue #27](https://github.com/exasol/error-reporting-python/issues/27)
     """
-    return Error(code, message, mitigations, parameters)
+    try:
+        return Error(code, message, mitigations, parameters)
+    except ValueError:
```

Comment:
is there only an invalid error code the reason for a ValueError or should we maybe define our own Exception for this case.

Reply:
As far as I could see the only explicitly propagated error is `ValueError` still you right it potentially could shadow an other `ValueError` not related to the Error Code Format.
I'll introduce a custom exception.